### PR TITLE
Add workflow for building live branches

### LINKED ISSUE #34523: Live Branches: Create webhook on build to notify Jetpack's Beta Download
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

A pull request kicks off a workflow to build an artifact of WooCommerce. After this has run, create a job that ping's Jetpack's Beta Download with metadata about the build.

Beta Download will grab the build and add it to its manifest.

Here is an example of Jetpack's job

https://github.com/Automattic/jetpack/blob/53d99a9ae0a67f86dfb401cad8b9b4525d7d8a74/.github/workflows/build.yml#L269-L281

Here is an example of Jetpack's manifest https://betadownload.jetpack.me/jetpack-branches.json

The beta download should be already configured to accept multiple plugins, see https://github.com/Automattic/jetpack-builder/pull/18 for background there. 

- [ ] Create hook to ping beta download
- [ ] Ensure builds are added to a WooCommerce manifest
- [ ] Ensure a build is removed from the manifest once the pull request is merged or closed

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR builds a zip from each PR for WooCommerce core and submits it to the Jetpack Beta Downloader manifest.

Closes #34523 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Create a new PR based on this PR.
2. Verify that the "Build Live Branch" workflow completes successfully. It should contain a single artifact zip.
3. Wait a few minutes for the Jetpack Beta Downloader to slurp up the artifact file, and then verify that your branch appears within [this json](https://betadownload.jetpack.me/woocommerce-branches.json).
4. Close your PR. Again, wait some time. Your branch should be removed from the json.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by psealock** [comment]
Nice to see you've got something populating the manifest. Do you know how branches are removed from the manifest once they are merged?

File: .github/workflows/scripts/get-plugin-version.js
Comment (psealock) [inline .github/workflows/scripts/get-plugin-version.js:4]: It feels like we should have a utility for this. Does this type of thing help? https://github.com/woocommerce/woocommerce/pull/34951

**Comment by jonathansadowski** [comment]
> Nice to see you've got something populating the manifest. Do you know how branches are removed from the manifest once they are merged?

I'm not entirely sure if there's some polling in place to check on active branches or if something else is handling this, but I tested this via #35579, and the branch was removed from the manifest a couple of minutes after I closed the PR.

File: .github/workflows/scripts/get-plugin-version.js
Comment (jonathansadowski) [reply .github/workflows/scripts/get-plugin-version.js:4]: Thanks. I initially tried to get it going with that, but I was unable. Do you think we should wait on #34951 to be merged and rework this or maybe we can revisit this and clean it up some after #34951 lands?

File: .github/workflows/pr-build-live-branch.yml
Comment (psealock) [inline .github/workflows/pr-build-live-branch.yml]: I don't think these types are required

> if no activity types are specified, the workflow runs when a pull request is opened or reopened or when the head branch of the pull request is updated

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

File: .github/workflows/pr-build-live-branch.yml
Comment (psealock) [inline .github/workflows/pr-build-live-branch.yml]: We don't need on push as well, it creates two jobs

image

**Review by psealock — COMMENTED** [review]
This is testing nicely, nice work. Just a couple comments on when the job runs

File: .github/workflows/pr-build-live-branch.yml
Comment (jonathansadowski) [reply .github/workflows/pr-build-live-branch.yml]: Thanks... I had these in place while I was testing for easier retriggering, but forgot to remove them. Updated!

Comment (psealock) [reply .github/workflows/pr-build-live-branch.yml]: Did you push? I'm not seeing this line updated

Comment (jonathansadowski) [reply .github/workflows/pr-build-live-branch.yml]: Doh, I had overwrote the change I made on GH's UI with a force-push to fix a failing test. Re-fixed now 😄

**Review by psealock — APPROVED** [review]
This looks great! Nice one. We can handle the `get-plugin-version` issue in a follow up 🚢

**Pull request merged**
